### PR TITLE
Don't really push silence to the ring buffer

### PR DESCRIPTION
**Pull request by padenot**
This implements your idea, only by removing code it turns out.

### CODE REVIEW

**Comment by padenot** [comment]
I've tested it on a few devices on my macbook 2018: built-in speakers/mic, an hd25 plugged in the TRRS socket(it doesn't have a mic), and a Sony WH-1000 xm3, that has a mic, via bluetooth, and trying all sorts of combinations and switches.

File: src/backend/mod.rs [hunk 1]
```diff
@@ -611,8 +608,6 @@ extern "C" fn audiounit_output_callback(
                     || stm.frames_read.load(Ordering::SeqCst) == 0)
             {
                 let silent_frames_to_push = input_frames_needed - buffered_input_frames;
-                let silent_samples_to_push = silent_frames_to_push * input_channels;
-                input_buffer_manager.push_silent_data(silent_samples_to_push);
```
Comment (ChunMinChang) [inline src/backend/mod.rs:615]: Can you add some comments here, or above `get_linear_data`, or `cubeb_log!` below to not that the silent data is appended silently in `get_linear_data`?

**Review by ChunMinChang — APPROVED** [review]
LGTM! 

r+ with some comments added.

File: src/backend/mod.rs (hunk 1 above)
Comment (padenot) [reply src/backend/mod.rs:615]: Yep did both.

**Comment by ChunMinChang** [comment]
Add a reference: This fixes [BMO 1629839](https://bugzilla.mozilla.org/show_bug.cgi?id=1629839).

**Pull request merged**